include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf.cpp

```diff
@@ -383,73 +383,42 @@ void
 TimeIntBDF<dim, Number>::get_velocities_and_times(std::vector<VectorType const *> & velocities,
                                                   std::vector<double> &             times) const
 {
-  /*
-   * the convective term is nonlinear, so we have to initialize the transport velocity
-   * and the discrete time instants that can be used for interpolation
-   *
-   *   time t
-   *  -------->   t_{n-2}   t_{n-1}   t_{n}     t_{n+1}
-   *  _______________|_________|________|___________|___________\
-   *                 |         |        |           |           /
-   *               sol[2]    sol[1]   sol[0]
-   *             times[2]  times[1]  times[0]
-   */
-  unsigned int current_order = this->order;
-  if(this->time_step_number <= this->order and this->param.start_with_low_order == true)
-  {
-    current_order = this->time_step_number;
-  }
-
-  AssertThrow(current_order > 0 and current_order <= this->order,
-              dealii::ExcMessage("Invalid parameter current_order"));
-
-  velocities.resize(current_order);
-  times.resize(current_order);
-
-  for(unsigned int i = 0; i < current_order; ++i)
-  {
-    velocities.at(i) = &get_velocity(i);
-    times.at(i)      = this->get_previous_time(i);
-  }
```

Comment:
This code is much more readable in my opinion than the current suggestion. I suggest to generalize the above code by just replacing `velocity` by a more general `vector` and a corresponding lambda `get_vector(i)`.

Reply:
[1f39cef](https://github.com/exadg/exadg/pull/458/commits/1f39cef8d688232b73b239620a45f409a169e862)